hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -173,7 +175,12 @@ private RecordIterator(Schema readerSchema, Schema writerSchema, byte[] content)
         this.totalRecords = this.dis.readInt();
       }
 
-      this.reader = new GenericDatumReader<>(writerSchema, readerSchema);
+      if (recordNeedsRewriteForExtendedAvroTypePromotion(writerSchema, readerSchema)) {
+        this.reader = new GenericDatumReader<>(writerSchema, writerSchema);
```

Comment:
line 217, we rewrite old record with new schema.
is it possible to support column reorder also ?